scrapy_zyte_api/_request_fingerprinter.py

```diff
@@ -0,0 +1,81 @@
+from typing import TYPE_CHECKING
+
+try:
+    from scrapy.utils.request import RequestFingerprinter  # NOQA
+except ImportError:
+    if not TYPE_CHECKING:
+        ScrapyZyteAPIRequestFingerprinter = None
+else:
```

Comment:
I'm thinking of ways if we could simplify the configuration and setup on this regard. Currently, users need to set the following if they have Scrapy 2.7+
- `REQUEST_FINGERPRINTER_CLASS`
- `ZYTE_API_FALLBACK_REQUEST_FINGERPRINTER_CLASS`

If users are using Scrapy versions below this, they won't need to set it. But if they do upgrade in the future, it's another step that they need to consider. Probably even forget about it.

What do you think about setting `REQUEST_FINGERPRINTER_CLASS="scrapy_zyte_api.ScrapyZyteAPIRequestFingerprinter"` by default regardless of what Scrapy version they use?

But in this case, `ScrapyZyteAPIRequestFingerprinter` automatically detects which Scrapy version the user is using and changes its behavior accordingly. So when users do upgrade Scrapy versions in the future, it's one less setting modification that they need to worry about.

One downside here though is that such convenience could remove the user's awareness that the Fingerprinting methodology has changed.

Reply:
Setting the setting on older Scrapy versions is actually already possible. We could update the documentation not to mention the 2.7 requirement, or to clarify that the setting should be set, but it will only be effective on Scrapy 2.7 and later.

Reply:
I see, that's awesome! 

+1 on this approach since the user can just set and forget it. No need to revisit and update the fingerprinter classes later on when they upgrade Scrapy.